src/main/java/dataMgmt/models/ProjectMetadata.java

```diff
@@ -54,26 +47,35 @@ public void setFiles(FileMetadata[] files) {
         this.files = files;
     }
 
+    public long getFileVersion() {
+        return fileVersion;
+    }
+
+    public void setFileVersion(long fileVersion) {
+        this.fileVersion = fileVersion;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
+        if (!(o instanceof ProjectMetadata)) return false;
 
-        ProjectMetadata metadata = (ProjectMetadata) o;
+        ProjectMetadata that = (ProjectMetadata) o;
 
-        if (getProjectId() != metadata.getProjectId()) return false;
-        if (getName() != null ? !getName().equals(metadata.getName()) : metadata.getName() != null) return false;
-        if (getOwner() != null ? !getOwner().equals(metadata.getOwner()) : metadata.getOwner() != null) return false;
-        return Arrays.equals(getFiles(), metadata.getFiles());
+        if (projectID != that.projectID) return false;
+        if (fileVersion != that.fileVersion) return false;
+        if (name != null ? !name.equals(that.name) : that.name != null) return false;
+        // Probably incorrect - comparing Object[] arrays with Arrays.equals
+        return Arrays.equals(files, that.files);
```

Comment:
If you want to see if the files arrays are the same, you'll need a FileMetadata.equals defined. I believe Arrays.equals calls the .equals function on each object.